docs/docs/routing.md

```diff
@@ -2,6 +2,10 @@
 title: Routing
 ---
 
+Part of what makes Gatsby sites so fast is that a lot of the work is done at build time and the running site is using mostly [static content](/docs/adding-app-and-website-functionality/#static-pages). Specifically, each page is its own HTML file that's generated at build time. During that process, Gatsby creates paths to access those files, handling routing for you. Navigating in a Gatsby app requires understanding what those paths are and how they're generated.
+
+Alternatively, your application may include functionality that cannot be handled at build time or through [rehydration](/docs/adding-app-and-website-functionality/#how-hydration-makes-apps-possible). This includes things like authentication, transactions, or form submission. To handle those pages, you can make use of [client-only routes](/docs/client-only-routes-and-user-authentication) using `@reach/router` which is built into Gatsby.
```

Comment:
I'm a little unsure about using transactions and form submissions as examples, since the static pages are rehydrated they can still technically fire off requests to other services client-side. 

Maybe using an example like: 
```suggestion
Alternatively, your application may include functionality that cannot be handled at build time or through [rehydration](/docs/adding-app-and-website-functionality/#how-hydration-makes-apps-possible). This includes things like authentication or routes for dynamic content like products in a database (e.g. `/products/:productId`). To handle those pages, you can make use of [client-only routes](/docs/client-only-routes-and-user-authentication) using `@reach/router` which is built into Gatsby.
```

Not sure what's best 🤔